src/browser/api/webcontents.ts

```diff
@@ -0,0 +1,90 @@
+import * as url from 'url';
+
+
+export function executeJavascript(webContents: Electron.WebContents, code: string, callback: (e: any, result: any) => void): void {
+    webContents.executeJavaScript(code, true, (result) => {
+        callback(undefined, result);
+    });
+}
+
+export function getInfo(webContents: Electron.WebContents) {
+    return {
+        canNavigateBack: webContents.canGoBack(),
+        canNavigateForward: webContents.canGoForward(),
+        title: webContents.getTitle(),
+        url: webContents.getURL()
+    };
+}
+
+export function getAbsolutePath(webContents: Electron.WebContents, path: string) {
+    const windowURL = webContents.getURL();
+
+    return  url.resolve(windowURL, path);
+}
+
+export function navigate (webContents: Electron.WebContents, url: string) {
+    // todo: replace everything here with "return webContents.loadURL(url)" once we get to electron 5.*
+    // reason: starting electron v5, loadUrl returns a promise that resolves according to the same logic we apply here
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.loadURL(url);
+    return navigationEnd;
+}
+
+export async function navigateBack (webContents: Electron.WebContents) {
+    if (!webContents.canGoBack()) {
+        throw new Error('Cannot navigate back');
+    }
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.goBack();
+    return navigationEnd;
+}
+
+export async function navigateForward (webContents: Electron.WebContents) {
+    if (!webContents.canGoForward()) {
+        const error = new Error('Cannot navigate forward');
+        throw error;
+    }
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.goForward();
+    return navigationEnd;
+}
+
+export function getZoomLevel(webContents: Electron.WebContents, callback: (zoomLevel: number) => void) {
+    webContents.getZoomLevel(callback);
+}
+
+export function reload(webContents: Electron.WebContents, ignoreCache: boolean = false) {
+    if (!ignoreCache) {
+        webContents.reload();
+    } else {
+        webContents.reloadIgnoringCache();
+    }
+}
+
+export function setZoomLevel(webContents: Electron.WebContents, level: number) {
+    // webContents.setZoomLevel(level); // zooms all windows loaded from same domain
+    webContents.send('zoom', { level }); // zoom just this window
+}
+
+export function stopNavigation(webContents: Electron.WebContents) {
+    webContents.stop();
+}
+
+function createNavigationEndPromise(webContents: Electron.WebContents): Promise<void> {
+    return new Promise((resolve, reject) => {
+        const chromeErrCodesLink = 'https://cs.chromium.org/chromium/src/net/base/net_error_list.h';
+        // tslint:disable-next-line:no-empty
+        let didSucceed = () => { };
```

Comment:
isn't `didSucceed` always gonna be what's being assigned on line #83?

Reply:
Yeah, just being paranoid, will fix